### DISCUSSION TITLE
Possible regression with useInfiniteQuery when deleting last element from a page

### DISCUSSION THREAD

**Discussion by wprk** [TanStack/query#2259] [Q&A]
I've got the same issue described [here](https://github.com/tannerlinsley/react-query/issues/1115) but I'm using version `3.13.10` and that issue was meant to be fixed in `2.23.1` - Is there a chance there has been a regression?

I've taken a look at the code [here](https://github.com/tannerlinsley/react-query/blob/master/src/core/infiniteQueryBehavior.ts) but It's changed quite a but since version `2.23.1` so can't work out whether it's a bug on my end or with the library. Any help would be apprieciated.

**Comment by TkDodo** [comment]
I can take a look if you provide a minimal runnable codesandbox example

**Reply by wprk** [reply]
Sure will do, I'm in Australia so it's almost midnight now but I'll do it first thing tomorrow. Thank you :)

**Comment by wprk** [comment]
Hi @TkDodo - I've created a sandbox demonstrating the issue.

You can replicate the issue by clicking the heart button on each prroduct in the row to remove it until you get to the last product in that row (page) and then it errors.

https://codesandbox.io/s/condescending-bash-04mex?file=/src/components/WishlistContainer.js

Thanks again for your help

**Reply by TkDodo** [reply]
> You can replicate the issue by clicking the heart button on each prroduct in the row to remove it until you get to the last product in that row (page) and then it errors.

okay I've been clicking that button like 10 times now and more images keep popping up. After they are all gone, there is no error visible. Please make the example more minimal and easy to reproduce - I'm not going through hundrets of lines of code. By doing that, you might narrow it down yourself and find the issue :)
